### PR TITLE
Migration of the RETORCH Annotations to Github close #6

### LINKED ISSUE #6: Migrate retorch annotations
Migrate the annotations. Once merged, I will release v1.1.0 to Maven Central

### PR DESCRIPTION
**Pull request by augustocristian**
- Created classes with the custom annotations
- Improved grammar and spell of the original comments

### CODE REVIEW

**Comment by augustocristian** [comment]
Its necesary to remove Class2 and its test of the repository prior to merge this pull request? @javiertuya

File: retorch-annotations/src/main/java/giis/retorch/annotations/AccessMode.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package giis.retorch.annotations;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+//Copyright 2020, Cristian Augusto, All rights reserved.
```
Comment (javiertuya) [inline retorch-annotations/src/main/java/giis/retorch/annotations/AccessMode.java:7]: Remove, we haven't a header yet, and copyright is uniovi

File: retorch-annotations/src/main/java/giis/retorch/annotations/AccessMode.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ * The RETORCH custom annotation for Access Mode has the following attributes:
```
Comment (javiertuya) [inline retorch-annotations/src/main/java/giis/retorch/annotations/AccessMode.java:10]: This comment does not provides additional useful information, see https://github.com/giis-uniovi/.github/blob/main/profile/CONTRIBUTING.md.
Comment should describe what is an access mode

File: retorch-annotations/src/main/java/giis/retorch/annotations/AccessMode.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * resID: A String representing the resource identifier, with a default value of "None."
+ * sharing: A Boolean indicating if sharing is enabled, with a default value of false.
+ * concurrency: An Integer specifying the concurrency level, with a default value of -1.
+ * accessMode: A String representing the access mode, with a default value of "NOASSIGNED"
```
Comment (javiertuya) [inline retorch-annotations/src/main/java/giis/retorch/annotations/AccessMode.java:14]: Similar comment than previous line. In this case, a good informative comment saying what is each attribute would be useful to the developer if the comments are placed at each attribute (to see their meaning when they write)

File: retorch-annotations/src/main/java/giis/retorch/annotations/AccessMode.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+public @interface AccessMode {
+    String resID() default "None";
+
+    boolean sharing() default false;
+
+    int concurrency() default -1;
+
+    String accessMode() default "NOASSIGNED";
+
+
```
Comment (javiertuya) [inline retorch-annotations/src/main/java/giis/retorch/annotations/AccessMode.java:28]: Avoid unnecesary spacing

**Review by javiertuya — CHANGES_REQUESTED** [review]
There are no tests, that make unused many things in the project  In addition to the comments placed in the annotations package:

Review all files in the project (both parent and module) to remove everything that is not needed (poms, workflows, remaining classes...).

**Comment by javiertuya** [comment]
@augustocristian Please, do not mention for every comment that is resolved. Simply mark it as resolved. When everything is resolved, request review. At this moment I will receive a TODO to be aware that I must start reviewing

**Comment by augustocristian** [comment]
> @augustocristian Please, do not mention for every comment that is resolved. Simply mark it as resolved. When everything is resolved, request review. At this moment I will receive a TODO to be aware that I must start reviewing

I apologize for any confusion. I didn't expect the commits to also notify the user if they are mentioned in them. Regarding the workflows, I have one question:
If I remove or deactivate workflows that are not used (e.g., test-java), the pull request doesn't pass because it's required. Am I misunderstanding what to do?
The idea is to have two minimal pom.xml files, each containing only the Maven stages that are useful. For example, the retorchanotations pom.xml should not have any dependencies related to testing, such as jacoco or logging.

**Comment by javiertuya** [comment]
> If I remove or deactivate workflows that are not used (e.g., test-java), the pull request doesn't pass because it's required. Am I misunderstanding what to do?
> The idea is to have two minimal pom.xml files, each containing only the Maven stages that are useful. For example, the retorchanotations pom.xml should not have any dependencies related to testing, such as jacoco or logging.

@augustocristian The test workflow should be kept, but only to compile (you may rename to build.yml if you prefer, but when other modules be included you would need to rename again). So, there is no problem, the test workflow triggers sonarqube, and everything is as before.

Regarding poms, exactly, testing/reporting and logging dependencies should be removed

**Comment by augustocristian** [comment]
I have noticed another requested change that is marked as unresolved when all is solved-marked :( . I have modified the workflows to remove the required artifacts and changed the Maven command to compile.

**Review by javiertuya — APPROVED** [review]

**Comment by javiertuya** [comment]
@augustocristian FYI, Artifact published in maven central (version 1.1.0)

**Pull request merged**